docs/enhancements/metricsConfig_redesign.md

```diff
@@ -0,0 +1,169 @@
+  - There's also the caveat of parsing through fixed and optional labels - ensuring there's no overlap and maintaining the order for certain metrics which do not accept map of label-value while observing, are some complications that currently exist and probably will not be fixed as a part of this enhancement.
+
+
+### Goals
+- Make all Hive metrics optional. By this design, no metric is to be reported by default unless it is explicitly configured in HiveConfig.
+- Allow configuring additional labels, duration based threshold and expression matching support for all metrics that are related to cluster deployment, via HiveConfig.
+- Enforce naming convention for duration based metrics - all metrics that report duration and allow customization for duration-based threshold must be named `*_seconds`.
+- Move away from throwing a panic if metricsConfig fails validation, update HiveConfig.status.Conditions instead
+
+### Non-Goals
+- Non-cluster deployment related metrics - like metrics reported by HiveOperator, or metrics that do not have the related cluster deployment available while reporting (for ex, some syncset, selectorsyncset and clusterpool metrics) will not support additional label and expression matching. 
+- Adding labels only refers to the labels propagated from the Cluster Deployment, any other strings as labels cannot be configured this way.
+
+
+## Proposal
+Allow HiveConfig.Spec.metricsConfig to look like
+```
+HiveConfig:
+  spec:
+    metricsConfig::
+    - metricsToReport:
+      - hive_foo_counter
+      - hive_foo_gauge
+      minimumDuration: 10m
+      clusterDeploymentLabelSelector:
+        matchLabels:
+          hive.openshift.io/aro-snowflake: "true"
+        matchExpressions:
+        - key: hive.openshift.io/limited-support
+          operator: NotIn
+          values: 
+          - "true"
+        - key: hive.openshift.io/limited-support
+          operator: NotExists
+      additionalClusterDeploymentLabels:
+        prom_label_name: hive.openshift.io/cd-label-key
+   - metricsToReport:
+     - hive_foo_histogram
+```
+### metricsToReport
+We would now default to not reporting any metrics unless they have been explicitly listed out under `metricsToReport`.
+There needs to be a non-empty list of hive metrics following an entry for `metricsToReport`, and there can be optional entries for customizations `minimumDuration`, `clusterDeploymentLabelSelector` and `additionalClusterDeploymentLabels`.
+All customizations listed for an entry of `metricsToReport` will apply to all the metrics provided in the corresponding list. This allows for grouping filters for a list of relevant metrics.
+Multiple entries for `metricsToReport` are allowed.
+Implementation will be adapting the existing setup for optional metrics. However, instead of a shorthand CamelCase key that is used for current `hiveconfig.spec.metricsConfig.metricsWithDuration`, we would now need the full metric name listed under `metricsToReport`.
```

Comment:
```suggestion
Implementation will be adapting the existing setup for optional metrics. However, instead of a shorthand camelCase key that is used for current `hiveconfig.spec.metricsConfig.metricsWithDuration`, we would now need the full metric name listed under `metricsToReport`.
```